dfttk/workflows.py

```diff
@@ -17,13 +17,172 @@
+
+    Args:
+        path (str): path to the folder containing the error folders.
+        error_folders_old (list[str]): list of error folders to rename.
+        error_folders_count (int): count of error folders that was moved using move_error_folders.
+    """
+    for error_folder_old in error_folders_old:
+        error_folder_new = error_folder_old.copy()
+        index = int(error_folder_new[1])
+        error_folder_new[1] = str(index + error_folders_count)
+        error_folder_new = ".".join(error_folder_new)
+        error_folder_old = ".".join(error_folder_old)
+
+        src_path = os.path.join(path, error_folder_old)
+        dest_path = os.path.join(path, error_folder_new)
+        subprocess.run(["mv", src_path, dest_path], check=True)
+
+
+def process_error_folders(path: str, error_folders_count: int) -> None:
+    """processes the error folders in the path by renaming and moving them.
+
+    Args:
+        path (str): path to the folder containing the error folders.
+        error_folders_count (int): count of error folders that was moved using move_error_folders.
+    """
+    error_folders_old = [f for f in os.listdir(path) if f.startswith("error.")]
+    error_folders_old = [f.split(".") for f in error_folders_old]
+    error_folders_old.reverse()
+    rename_error_folders(path, error_folders_old, error_folders_count)
+
+    error_temp_path = os.path.join(path, "error_temp")
+    error_temp_folders = [f for f in os.listdir(error_temp_path)]
+    move_error_folders(error_temp_path, path, error_temp_folders)
+
+    subprocess.run(["rm", "-r", error_temp_path])
+
+
+def handle_custodian_json(path: str) -> None:
+    """Merges the custodian.json and custodian_old.json files in the path.
+
+    Args:
+        path (str): path to the folder containing the custodian.json and custodian_old.json files.
+    """
+    custodian_json_path = os.path.join(path, "custodian.json")
+    custodian_old_json_path = os.path.join(path, "custodian_old.json")
+    if os.path.isfile(custodian_old_json_path):
+        with open(custodian_json_path, "r") as f:
+            custodian_json = json.load(f)
+        with open(custodian_old_json_path, "r") as f:
+            custodian_old_json = json.load(f)
+
+        custodian_json[0].update({"restart": True})
+        combined_custodian_json = custodian_old_json + custodian_json
+        with open(custodian_json_path, "w") as f:
+            json.dump(combined_custodian_json, f, indent=4)
+
+        os.remove(custodian_old_json_path)
+
+#TODO: Find a way to deal with custodian mistakenly naming indexing two error folders the same name
+def gzip_error_folders(path: str, error_folders: list[str]) -> None:
+    """Gzips the error folders in the path.
+
+    Args:
+        path (str): path to the folder containing the error folders.
+        error_folders (list[str]): list of error folders to gzip.
+    """
+    for error_folder in error_folders:
+        if not error_folder.endswith(".gz"):
```

Comment:
why would an `error_folder` end with `.gz`? This seems like you may need to correct how the error folders are collected.

Reply:
I think the custodian has an error where it zips some error folders to have .gz, but it doesn't do it for all of them. I will remove this function for now.